buildlib/tools/git_ready.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# See file LICENSE for terms.
+#
+
+source ./buildlib/tools/codestyle.sh
+
+git_commit() {
+    if ! git diff-index --quiet HEAD; then
+        git add -p
+        # returns 1 if cached index is not empty
+        git diff --cached --exit-code || git commit -m "$title"
+        git reset --hard
+    fi
+}
+
+ucx=https://github.com/openucx/ucx.git
+upstream=$(git remote -v  | grep -P "[\t ]${ucx}[\t ].*fetch" | cut -f 1 | head -n 1)
+base=$(git merge-base "$upstream"/master HEAD)
+
+if ! git diff-index --quiet HEAD; then
+    echo "error: tree not clean"
+    exit 1
+fi
+
+codestyle_check_commit_title "$base"..HEAD
+if [[ $? -ne 0 ]]
+then
+    echo "error: fix commit title"
+    exit 1
+fi
+
+# Indent
+module load dev/llvm-ucx || :
+git clang-format --diff "$base" HEAD | patch -p1
+module unload dev/llvm-ucx || :
+
+git_commit
+
+# Codespell
+codestyle_check_spell --write-changes || :
+
+git_commit
```

Comment:
why have git commit both here and in line 39?

Reply:
merged both commits in one